tests/test-amort.c

```diff
@@ -324,7 +325,15 @@ main (int argc, char *argv[])
   numinputs = ftell (inputfile) / sizeof (inputtype);
   rewind (inputfile);
   inputs = malloc (numinputs * sizeof (inputtype));
-  fread (inputs, sizeof (inputtype), numinputs, inputfile);
+
+  r = fread (inputs, sizeof (inputtype), numinputs, inputfile);
+  if (ferror (inputfile))
```

Comment:
I suspect you will also want to check that r == numinputs too.  It also removes the need for the unused attribute above.